modules/consensus/subscribe.go

```diff
@@ -169,12 +172,17 @@ func (cs *ConsensusSet) managedInitializeSubscribe(subscriber modules.ConsensusS
 		cs.mu.RLock()
 		err = cs.db.View(func(tx *bolt.Tx) error {
 			for i := 0; i < 100 && exists; i++ {
-				cc, err := cs.computeConsensusChange(tx, entry)
-				if err != nil {
-					return err
+				select {
+				case <-cancel:
+					return errors.New("aborting managedInitializeSubscribe")
```

Comment:
ah, I see. Hmm.
We don't have a clear convention for this because `StopChan` usually just causes a goroutine to exit, not a blocking call to return early. The only other instance I could find is in the renter:

```go
// wait for the download to complete and return the data
select {
case <-d.downloadFinished:
	return buf.Bytes(), d.Err()
case <-r.tg.StopChan():
	return nil, errors.New("chunk download interrupted by shutdown")
case <-time.After(chunkDownloadTimeout):
	return nil, errors.New("chunk download timed out")
}
```
so returning an error might be okay after all. Still, I think we'd at least want it to be an exported variable (e.g. `siasync.ErrStopped`) so that the caller can check for it explicitly.